prusaerrors/sl1/codes.py

```diff
@@ -10,15 +10,72 @@
 
 import builtins
 from pathlib import Path
+from re import compile as re_compile
 
-from prusaerrors.shared.codes import unique_codes, Codes, yaml_codes
+import yaml
+
+from prusaerrors.shared.codes import Category, Code, Printer, unique_codes, Codes
+from prusaerrors.sl1 import PRINTER_MODEL_PATH
 
 if "_" not in vars(builtins):
 
     def _(value):
         return value
 
 
+def yaml_codes(src_path: Path):
+    """
+    Add code definitions from YAML source
+    """
+
+    def decor(cls):
+        with src_path.open("r") as src_file:
+            data = yaml.safe_load(src_file)
+        assert "Errors" in data
+
+        printer = Printer.SL1
+        model = [x.name for x in Path(str(PRINTER_MODEL_PATH)).iterdir() if x.is_file()]
```

Comment:
I think PRINTER_MODEL_PATH can be defined as Path to avoid conversions. Currently it is a string, so at last the `str` could be removed.

Reply:
fixed